### DISCUSSION TITLE
BUG: searching

### DISCUSSION THREAD

**Discussion by jtnord** [community/community#40622] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

HI the "new" search experience contains a couple of bugs.

1. it does not support `filename` which is generated from the `advanced search page` meaning the advanced search page generates invalid queries when looking for specific filenames.

2. when searching for a specific file (path:/foo.txt) you are returned files that are not in the root and you have to use `path:/ path:**/foo.txt)  this is counter intuitive why would `/` not include sub directories `/foo.txt` would)  I could understand if I used just `foo.txt` without any path.

**Comment by jtnord** [comment]
> you are returned files that are not in the root and you have to use path:/ path:**/foo.txt)

I was lulled into a false sense - this does not work correctly either (contains false positive matches).  
https://github.com/search?q=user%3Ajenkinsci+path%3A**%2Fbuild.gradle+path%3A%2F&type=code has matches that are not in the root despite having less matches than https://github.com/search?q=user%3Ajenkinsci+path%3A%2Fbuild.gradle&type=code

**Comment by ossanna16** [comment]
Hi there @jtnord and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
